### ISSUE TITLE
Finding wrong python3 library on systems with multiple python versions

### ISSUE THREAD

**Issue by nightlark** [GMLC-TDC/HELICS#659]
**Describe the bug**
Recently there were issues building the python3 interface on a system, resulting in linking errors against a static python library saying to compile with -fPIC. The problem was found to be partly due to multiple python installs on a system installed to different locations. The main version, python 3.6 was installed in /usr/bin with the shared library in /usr/lib64. However, there was also a python 3.5 install in /usr/local/bin and a static library in /usr/local/lib. The CMake script was picking up the python 3.6 binary, and the python 3.5 static library.

**What is the expected behavior?**
CMake should have picked up the python library that goes with the 3.6 binary.

**To Reproduce**
Get a system with two different versions of python, one installed in /usr and the other in /usr/local. It might be important for the one in /usr/local to have the static version of the python library installed.

**Environment**
 - Operating System: Mac | Linux
 - Language Extension: Python
 - HELICS version: all

**Additional context and information**
This might be an issue with the FindPython*.cmake script that is included with CMake. One work around if anyone encounters a similar problem is to set the PYTHON_LIBRARY CMake variable to the full path to the right python library.

A one line bash command for getting the full path to the python library corresponding to the main python binary (that works with module load) is:
`python3 -c "from distutils import sysconfig; print(sysconfig.get_config_var('DESTDIRS').split(' ',2)[1] + '/' + sysconfig.get_config_var('INSTSONAME'))"`

Possible fix, add lib64 here:
https://gitlab.kitware.com/cmake/cmake/blob/352baee2074d12637c65e3113f560bbf98d0640a/Modules/FindPython/Support.cmake#L532

**Comment by kdheepak** [comment]
Can you paste the output of the CMake error as well? If others search for it they'll find this thread.

**Comment by nightlark** [comment]
There isn't CMake error, the only indication from CMake is if the version in the Python library name doesn't match the "suitable version" number. The output might look something like this:
```
Found PythonLibs: /usr/local/lib/libpython3.5m.a (found suitable version "3.6.6", minimum required is "3")
```

Also, when building, there will be  a large number of errors similar to this when linking  the C shared module _helics.so for the Python module:
```
/usr/bin/ld: /usr/local/lib/libpython3.5m.a(abstract.o): relocation R_X86_64_32S against symbol `_Py_NotImplementedStruct' can not be used when making a shared objec
t; recompile with -fPIC
```

**Comment by phlptp** [comment]
I think this was addressed by #818, but @kdheepak needs to test it more to be sure,

**Comment by nightlark** [comment]
#986 fixes the issues with mismatched Python interpreter and library versions being found using the old find method.
